tests/testthat/test-pivot_wider_pknca_results.R

```diff
@@ -0,0 +1,165 @@
+    original_vals <- original_data %>%
+      formatters::var_labels_remove() %>%
+      filter(PPTESTCD == pptestcd) %>%
+      pull(PPSTRES)
+    result_vals <- reshaped_data %>%
+      formatters::var_labels_remove() %>%
+      pull(param_col)
+
+    have_similar_vals <- all(round(result_vals, 3) %in% round(original_vals, 3))
+    have_same_length <- length(result_vals) == length(original_vals)
+    if (!(have_similar_vals && have_same_length)) {
+      stop(paste("Mismatch in values for parameter:", param_col))
+    }
+  }
+
+  # If all checks pass, return TRUE
+  TRUE
+}
+
+
+describe("pivot_wider_pknca_results", {
+
+  it("produces a data.frame with expected format when only reshaping main intervals", {
+    res_only_main <- TEST_PKNCA_RES
+    res_only_main$result <- TEST_PKNCA_RES$result  %>%
+      filter(type_interval == "main")
+    pwres_only_main <- pivot_wider_pknca_results(res_only_main)
+    expect_s3_class(pwres_only_main, "data.frame")
+
+    # Check that the result contains expected columns
+    expected_columns <- c(
+      "USUBJID", "start", "end", "DOSNO", "AFRLT", "ARRLT",
+      "NFRLT", "NRRLT", "ADOSE", "ROUTE", "ADOSEDUR",
+      "CMAX[ng/mL]", "TMAX[hr]", "TLST[hr]", "LAMZ[1/hr]",
+      "R2", "R2ADJ", "LAMZLL[hr]",
+      "LAMZNPT[count]", "CLSTP[ng/mL]", "LAMZHL[hr]",
+      "LAMZSPN", "LAMZIX",
+      "LAMZMTD", "Exclude"
+    )
+    expect_true(all(expected_columns %in% colnames(pwres_only_main)))
+
+    # Check number of rows is the expected (each dose has 1 row in the pivoted data)
+    expect_true(nrow(pwres_only_main) == nrow(res_only_main$data$dose$data))
+
+    # Check parameter values match the ones in the PKNCA results object
+    expect_true(.validate_pknca_params(res_only_main, pwres_only_main))
+  })
+
+  it("reshapes PKNCA results correctly when also considering AUC intervals", {
+    # Check that the result is a data frame
+    expect_s3_class(pwres, "data.frame")
+
+    # Check that the result contains expected columns
+    expected_columns <- c(
+      "USUBJID", "start", "end", "DOSNO", "AFRLT", "ARRLT",
+      "NFRLT", "NRRLT", "ADOSE", "ROUTE", "ADOSEDUR",
+      "CMAX[ng/mL]", "TMAX[hr]", "TLST[hr]", "LAMZ[1/hr]",
+      "R2", "R2ADJ", "LAMZLL[hr]",
+      "LAMZNPT[count]", "CLSTP[ng/mL]", "LAMZHL[hr]",
+      "LAMZSPN", "LAMZIX", "LAMZMTD", "Exclude",
+      "AUCINT_0-2[hr*ng/mL]", "AUCINT_2-4[hr*ng/mL]"
+    )
+    expect_true(all(expected_columns %in% colnames(pwres)))
+
+    expect_true(.validate_pknca_params(TEST_PKNCA_RES, pwres))
+  })
+
+  it("rounds numeric values to three decimals", {
+    expected_num_param_cols <- c(
+      "CMAX[ng/mL]", "TMAX[hr]", "TLST[hr]", "LAMZ[1/hr]",
+      "R2", "R2ADJ", "LAMZLL[hr]", "LAMZNPT[count]", "LAMZSPN",
+      "AUCINT_0-2[hr*ng/mL]", "AUCINT_2-4[hr*ng/mL]"
+    )
+    rounded_values <- pwres[, expected_num_param_cols]
+    expect_true(all(apply(rounded_values, 2,
+                          function(x) all(na.omit(abs(x - round(x, 3))) < 1e-9))))
```

Comment:
issue: This should be part of the setup. No need to split the setup in two parts, makes the check itself seem more complicated than it is.

```suggestion
    rounded_values <- pwres[, expected_num_param_cols] |>
      apply(2, function(x) all(na.omit(abs(x - round(x, 3))) < 1e-9))
    expect_true(all(rounded_values))
```

Reply:
@m-kolomanski  I refactored to this, hopefully looks more clear. Let me know if not!

```
  it("rounds numeric values to three decimals", {
    expected_num_param_cols <- c(
      "CMAX[ng/mL]", "TMAX[hr]", "TLST[hr]", "LAMZ[1/hr]",
      "R2", "R2ADJ", "LAMZLL[hr]", "LAMZNPT[count]", "LAMZSPN",
      "AUCINT_0-2[hr*ng/mL]", "AUCINT_2-4[hr*ng/mL]"
    )
    actual_values <- pwres[, expected_num_param_cols]
    expected_values <- round(actual_values, 3)
    diff_values <- actual_values - expected_values
    
    all_non_nan_values_are_rounded <- all(is.na(diff_values) | diff_values == 0)
    expect_true(all_non_nan_values_are_rounded)
  })
```